src/typeahead/typeahead.ts

```diff
@@ -271,6 +270,7 @@ export class NgbTypeahead implements ControlValueAccessor,
           event.preventDefault();
           this._resubscribeTypeahead.next(null);
           this.dismissPopup();
+          this._writeInputValue(this._inputValueBackup);
```

Comment:
I don't get the fix, why do it only on ESC and not on document click for instance?
Also I think `dismissPopup` can be called manually from code, will it work?